jpype/_core.py

```diff
@@ -104,23 +113,48 @@ def isJVMStarted():
     return _jpype.isStarted()
 
 
-def _hasClassPath(args):
+def _hasClassPath(args: typing.Tuple[_PathOrStr, ...]) -> bool:
     for i in args:
-        if i.startswith('-Djava.class.path'):
+        if isinstance(i, str) and i.startswith('-Djava.class.path'):
             return True
     return False
 
 
-def _handleClassPath(clsList):
+def _handleClassPath(
+    classpath: typing.Union[
+        _PathOrStr,
+        typing.Tuple[_PathOrStr, ...]
```

Comment:
I think we can also now just write tuple[_PathOrStr], what is the eclipse used for here?

Reply:
For tuple type you need the ellipsis if it is to be variable length (https://docs.python.org/3/library/typing.html#typing.Tuple). ``Tuple[_PathOrStr]`` is a length 1 tuple.

Reply:
Note, this is confusing because for ``List`` the types provided indicate variable length. So for example ``List[_PathOrStr]`` is variable length, not length 1. It is a bit surprising to me that this is the case with typing, but it is one of those things :shrug: 

https://docs.python.org/3/library/typing.html#typing.List